pkg/network/tracer/tracer.go

```diff
@@ -394,46 +402,73 @@ func (t *Tracer) Stop() {
 }
 
 // GetActiveConnections returns the delta for connection info from the last time it was called with the same clientID
-func (t *Tracer) GetActiveConnections(clientID string) (*network.Connections, error) {
+func (t *Tracer) GetActiveConnections(clientID string, maxConnectionPerMessage int) (*network.Connections, bool, error) {
 	t.bufferLock.Lock()
 	defer t.bufferLock.Unlock()
-	log.Tracef("GetActiveConnections clientID=%s", clientID)
+	log.Tracef("GetActiveConnections clientID=%s maxConnectionPerMessage=%d", clientID, maxConnectionPerMessage)
 
-	t.ebpfTracer.FlushPending()
-	latestTime, err := t.getConnections(t.activeBuffer)
-	if err != nil {
-		return nil, fmt.Errorf("error retrieving connections: %s", err)
+	cnx, found := t.clientConnections[clientID]
+	if !found {
+		t.ebpfTracer.FlushPending()
+		latestTime, err := t.getConnections(t.activeBuffer)
+		if err != nil {
+			return nil, false, fmt.Errorf("error retrieving connections: %s", err)
+		}
+		t.clientConnections[clientID] = connections{
+			latestTime: latestTime,
+			active:     t.activeBuffer.Connections(),
+		}
```

Comment:
This will not copy the underlying buffer no?  So if getActiveConnections() gets called with another client, the current value in clientConnections[firstClient] will get clobbered